docs/antrea-network-policy.md

```diff
@@ -948,6 +955,71 @@ spec:
 The above example allows all Pods from Namespace "default" to connect to all "core-dns"
 Pods from Namespace "kube-system" on TCP port 53.
 
+## FQDN based filtering
+
+Antrea-native policy accepts a `fqdn` field to restrict egress access to the Fully
+Qualified Domain Names (FQDN) specified by exact name or wildcard matching patterns.
+
+The standard `Allow`, `Drop` and `Reject` actions apply to FQDN egress rules.
+
+An example FQDN based filtering could look like this:
+
+```yaml
+apiVersion: crd.antrea.io/v1alpha1
+kind: ClusterNetworkPolicy
+metadata:
+  name: cnp-fqdn-all-google
+spec:
+  priority: 1
+  appliedTo:
+  - podSelector:
+      matchLabels:
+        app: client
+  egress:
+  - action: Drop
+    to:
+      - fqdn: "*google.com"
+```
+The above example drops all traffic destined to any FQDN that matches the wildchard
+expression `*google.com` originating from any Pod with `app` set to `client` across
+any Namespace. This feature only works at the L3/L4 level. DNS response packets will
+only be dropped if the client's requested FQDN is matched by a FQDN policy rule but
+the datapath reconciliation for that rule fails.
+
+The FQDN based policy can also be applied to in-cluster based services.
+
+```text
+kubectl get svc -o wide -A
+NAMESPACE     NAME          TYPE          CLUSTER-IP      EXTERNAL-IP   PORT(S)                   AGE   SELECTOR
+default       db-svc        ClusterIP     10.108.21.84    <none>        443/TCP                   2d    app=db
+default       kubernetes    ClusterIP     10.96.0.1       <none>        443/TCP                   2d    <none>
+kube-system   antrea        ClusterIP     10.98.109.50    <none>        443/TCP                   3d    <app=antrea,component=antrea-controller>
+kube-system   kube-dns      ClusterIP     10.96.0.10      <none>        53/UDP,53,TCP,9153/TCP    3d    k8s-app=kube-dns
+```
+
+In the above, there is a db service in the default namespace of type `ClusterIP`. A
+ClusterNetorkPolicy can be defined as follows:
+
+```yaml
+apiVersion: crd.antrea.io/v1alpha1
+kind: ClusterNetworkPolicy
+metadata:
+  name: cnp-fqdn-db-svc
+spec:
+  priority: 1
+  appliedTo:
+  - podSelector:
+      matchLabels:
+        app: client
+  egress:
+  - action: Drop
+    to:
+      - fqdn: "db-svc.default.svc.cluster.local"
+```
+
+In the example, when a client Pod talks to `db-svc` via ClusterIP, the traffic is dropped.
```

Comment:
it might also be good to highlight the fact that this is a exact name matching policy ... unlike the wildcard example above..